### PR TITLE
Add new sample for video processing with insertable streams

### PR DESCRIPTION
**Pull request by dogben**
First draft, including the following UI options:

- MediaStream source selection:
  * Camera
  * Video
  * Peer connection (from camera)
- Optionally include the source video in the page
- Transform selection:
  * WebGL
  * Canvas2D
  * Drop frames at random
  * Delay all frames by 100ms
- Transformed stream destination selection:
  * Video
  * Peer connection (to video)

Includes a link to the [explainer](https://github.com/w3c/mediacapture-insertable-streams) and a note regarding the experimental nature of the API and how to enable it.

### CODE REVIEW

File: src/content/insertable-streams/video/css/main.css [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
```
Comment (KaptenJansson) [inline src/content/insertable-streams/video/css/main.css]: Current copyright year for new files.

File: src/content/insertable-streams/video/index.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
+<!DOCTYPE html>
+<!--
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
```
Comment (KaptenJansson) [inline src/content/insertable-streams/video/index.html]: Current copyright year for new files.

File: src/content/insertable-streams/video/index.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    <p>This sample shows how to perform processing on a video stream using the experimental
+        <a href="https://github.com/w3c/mediacapture-insertable-streams">insertable streams</a> API.
+        There are options for the source of the input stream, the destination of the output stream,
+        and the API used to transform the stream. There is also the option to duplicate the source
+        stream to a video element on the page, which may affect the source FPS.
+    </p>
+
+    <span id="outputVideo"></span>
+    <!--video id="sourceVideo" playsinline autoplay muted></video-->
```
Comment (KaptenJansson) [inline src/content/insertable-streams/video/index.html]: Remove commented out code.

File: src/content/insertable-streams/video/js/main.js [hunk 4]
```diff
@@ -0,0 +1,1205 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
```
Comment (KaptenJansson) [inline src/content/insertable-streams/video/js/main.js]: Current copyright year for new files.

File: src/content/insertable-streams/video/js/main.js [hunk 5]
```diff
@@ -0,0 +1,1205 @@
...
+  /**
+   * @param {!MediaStream} stream
+   */
+  async setMediaStream(stream) {}
+  /** Frees any resources used by this object. */
+  destroy() {}
+}
+
+/**
+ * Assebles a MediaStreamSource, FrameTransform, and MediaStreamSink together.
```
Comment (KaptenJansson) [inline src/content/insertable-streams/video/js/main.js]: Assemble

**Review by KaptenJansson — CHANGES_REQUESTED** [review]
Made some comments inline but one general remark, the main.js file is over 1200 lines, maybe split it up? I know it's just a demo/sample page but that file is a bit daunting and I think splitting it up into logical pieces would make sense here. Maybe split into libraries and an interface/api file? @alvestrand WDYT?

**Comment by dogben** [comment]
> ... the main.js file is over 1200 lines, maybe split it up? ...

Done. If there's a better way to include definitions across files for this repo, let me know.

File: src/content/insertable-streams/video/js/main.js [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+ */
+let FrameTransformFn; // eslint-disable-line no-unused-vars
+
+/**
+ * Creates a pair of MediaStreamTrackProcessor and MediaStreamTrackGenerator
+ * that applies transform to sourceTrack. This function is the core part of the
+ * sample, demonstrating how to use the new API.
+ * @param {!MediaStreamTrack} sourceTrack the video track to be transformed. The
+ *     track can be from any source, e.g. getUserMedia, RTCTrackEvent, or
+ * captureStream on HTMLMediaElement or HTMLCanvasElement.
```
Comment (KaptenJansson) [inline src/content/insertable-streams/video/js/main.js]: Match indentation above.

File: src/content/insertable-streams/video/js/main.js [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+    initPipeline();
+  } else {
+    updatePipelineSourceIfSet();
+  }
+}
+sourceSelector.oninput = updatePipelineSource;
+sourceVisibleCheckbox.oninput = () => {
+  console.log(`[UI] Changed source visibility: ${
+      sourceVisibleCheckbox.checked ? 'added' : 'removed'}`);
+  // pipeline?.getSource()?.setVisibility(sourceVisibleCheckbox.checked);
```
Comment (KaptenJansson) [inline src/content/insertable-streams/video/js/main.js]: Remove?

**Review by KaptenJansson — APPROVED** [review]
LGTM with nits. Really impressive!

**Pull request merged**